frontend/dockerfile/dockerfile_test.go

```diff
@@ -2061,9 +2476,11 @@ COPY sub/dir1 subdest6
 	require.NoError(t, err)
 	require.Equal(t, "foo-contents", string(dt))
 
-	dt, err = ioutil.ReadFile(filepath.Join(destDir, "subdest/dir1/dir2/foo"))
-	require.NoError(t, err)
-	require.Equal(t, "foo-contents", string(dt))
+	if isFileOp { // non-fileop implementation is historically buggy
```

Comment:
This is a behavior change, but I verified that this is also the behavior in legacy docker build so the previous testcase was not correct.